src/Microsoft.DotNet.Darc/DarcLib/GitHubClient.cs

```diff
@@ -660,7 +660,10 @@ public async Task<List<GitFile>> GetFilesAtCommitAsync(string repoUri, string co
 
         if (Cache != null)
         {
-            return await Cache.GetOrCreateAsync((treeItem.Path, treeItem.Sha), async (entry) =>
+            // We're adding the full path here because the eng/common files have the same relative path in the VMR
+            // and in product repos relative to the eng/common folder, and we don't want to get bad cache hits.
+            // Their full paths are different so this mitigates the problem
+            return await Cache.GetOrCreateAsync((path, treeItem.Path, treeItem.Sha), async (entry) =>
```

Comment:
```suggestion
            return await Cache.GetOrCreateAsync(($"{path}/{treeItem.Path}", treeItem.Sha), async (entry) =>
```
we could also do this, but I don't think we need the extra string interpolation, think the above is more efficient.